stata_kernel/stata_session.py

```diff
@@ -325,6 +329,9 @@ def expect(self, text, child, md5, text_to_exclude=None, display=True):
                 continue
             if match_index == 2:
                 g_path = [self.expect_graph(child, child.match.group(0))]
+                if g_path[0] is None:
```

Comment:
@simon-d-s Sorry I've been MIA. Can this be fixed by instead adding `^` at the start of `g_exp`? Instead of the logic here and the `if else` logic below, add `^` at the start of `g_exp` so that the messages for a file not existing and the file being saved are not confused?

Reply:
@mcaceresb No problem.
For Stata 17 it would work, but I cannot see how it would work for Stata 16.

Reply:
@simon-d-s I tested it out on [this branch](https://github.com/kylebarron/stata_kernel/pull/392/files) and it was working for me on 15 and 16. I don't have 17 though.

Reply:
@mcaceresb Sorry, I did not fully understand what you mean - with the code it's clear now.
Unfortunately, this will not work on Stata 17. As you can see in the following example, the beginning the beginnings of the messages for a file not existing and the file being saved are not confused look differ only by the `(`: 
```
. sysuse auto
(1978 automobile data)

. hist price
(bin=8, start=3291, width=1576.875)

. graph export foobar.png, replace
(file foobar.png not found)
file foobar.png saved as PNG format
```

Reply:
@simon-d-s Ah, I see what you mean. I'll try to review your proposed fix when I get a sec. These weeks are a bit full for me, though.